test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,320 @@
+		}
+	})
+
+})
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func getTunedProfile(tunedName string) *tunedv1.Tuned {
+	tunedList := &tunedv1.TunedList{}
+	var matchedTuned *tunedv1.Tuned
+
+	Eventually(func(g Gomega) {
+		g.Expect(testclient.DataPlaneClient.List(context.TODO(), tunedList, &client.ListOptions{
+			Namespace: components.NamespaceNodeTuningOperator,
+		})).To(Succeed())
+
+		g.Expect(len(tunedList.Items)).To(BeNumerically(">", 1))
+
+		for _, tuned := range tunedList.Items {
+			if tuned.Name == tunedName {
+				matchedTuned = &tuned
+				break
+			}
+		}
+
+	}).WithTimeout(time.Minute * 3).WithPolling(time.Second * 10).Should(Succeed())
+
+	if matchedTuned == nil {
+		return nil
+	}
+	return matchedTuned
+}
+
+func createTunedObject(deferMode string, kernel_shmmni string, np *hypershiftv1beta1.NodePool, name string) (*tunedv1.Tuned, error) {
+	tunedName := name
+	ns := components.NamespaceNodeTuningOperator
+	priority := uint64(19)
+	data := fmt.Sprintf(`
+	[main]
+	summary=Configuration changes profile inherited from performance created tuned
+	include=openshift-node-performance-performance
+	[sysctl]
+	kernel.shmmni=%s
+	`, kernel_shmmni)
+
+	// Create a Tuned object
+	tuned := &tunedv1.Tuned{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: tunedv1.SchemeGroupVersion.String(),
+			Kind:       "Tuned",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tunedName,
+			Namespace: ns,
+			Annotations: map[string]string{
+				"tuned.openshift.io/deferred": deferMode,
+			},
+		},
+		Spec: tunedv1.TunedSpec{
+			Profile: []tunedv1.TunedProfile{
+				{
+					Name: &tunedName,
+					Data: &data,
+				},
+			},
+			Recommend: []tunedv1.TunedRecommend{
+				{
+					MachineConfigLabels: map[string]string{"machineconfiguration.openshift.io/role": testutils.RoleWorkerCNF},
+					Priority:            &priority,
+					Profile:             &tunedName,
+				},
+			},
+		},
+	}
+
+	testlog.Infof("Creating Tuned Profile Patch: %s", name)
+	Expect(testclient.ControlPlaneClient.Create(context.TODO(), tuned)).To(Succeed())
+	return tuned, nil
+}
+
+func execSysctlOnWorkers(ctx context.Context, workerNodes []corev1.Node, sysctlMap map[string]string) {
+	var err error
+	var out []byte
+	isSNO := false
+	for _, node := range workerNodes {
+		for param, expected := range sysctlMap {
+			By(fmt.Sprintf("executing the command \"sysctl -n %s\"", param))
+			tunedCmd := []string{"/bin/sh", "-c", "chroot /host sysctl -a | grep shmmni"}
+			tunedPod := nodes.TunedForNode(&node, isSNO)
+			out, err = pods.WaitForPodOutput(context.TODO(), testclient.K8sClient, tunedPod, tunedCmd)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(strings.TrimSpace(strings.SplitN(string(out), "=", 2)[1])).Should(Equal(expected), "parameter %s value is not %s.", out, expected)
+		}
+	}
+}
+
+func verifyImmediateApplication(tuned *tunedv1.Tuned, workerCNFNodes []corev1.Node, kernel_shmmni string) {
+	time.Sleep(40 * time.Second)
```

Comment:
why this Sleep? can we wait (with a timeout) on some condition rather than just sleep for a while?

Reply:
Replaced with Eventually block with a condition.